src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,8 +1,26 @@
-export const TodoFilter = () => (
+import { StatusFilterValue } from '../../types/Todo';
+
+interface Props {
+  onStatusSelected: (value: StatusFilterValue) => void;
+  statusFilter: StatusFilterValue;
+  onQueryChange: (value: string) => void;
+  query: string;
+}
+
+export const TodoFilter: React.FC<Props> = ({
+  onStatusSelected,
+  statusFilter,
+  onQueryChange,
+  query,
+}) => (
   <form className="field has-addons">
     <p className="control">
       <span className="select">
-        <select data-cy="statusSelect">
+        <select
+          data-cy="statusSelect"
+          value={statusFilter}
+          onChange={e => onStatusSelected(e.target.value as StatusFilterValue)}
+        >
           <option value="all">All</option>
           <option value="active">Active</option>
           <option value="completed">Completed</option>
```

Comment:
When you have an enum you can get an array of values and map through it to create links